connector/profiler/src/main/scala/org/apache/spark/executor/profiler/ExecutorJVMProfiler.scala

```
@@ -0,0 +1,183 @@
+          if (threadpool != null) {
+            threadpool.shutdownNow()
+          }
+          if (inputStream != null) {
+            inputStream.close()
+          }
+          if (outputStream != null) {
+            outputStream.close()
+          }
+      }
+    }
+  }
+
+  private def writeChunk(lastChunk: Boolean): Unit = {
+    if (!writing) {
+      return
+    }
+    try {
+      // stop (pause) the profiler, dump the results and then resume. This is not ideal as we miss
+      // the events while the file is being dumped, but that is the only way to make sure that
+      // the chunk of data we are copying to dfs is in a consistent state.
+      profiler.get.execute(stopcmd)
+      profiler.get.execute(dumpcmd)
+      var remaining = inputStream.available()
+      if (!lastChunk) {
+        profiler.get.execute(resumecmd)
+      }
+      while (remaining > 0) {
+        val read = inputStream.read(dataBuffer, 0, math.min(remaining, UPLOAD_SIZE))
+        outputStream.write(dataBuffer, 0, read)
+        remaining -= read
+      }
+    } catch {
+      case e: IOException => logError("Exception occurred while writing some profiler output: ", e)
+      case e@(_: IllegalArgumentException | _: IllegalStateException) =>
+        logError("Some profiler output not written." +
+          " Exception occurred in profiler native code: ", e)
+      case e: Exception => logError("Some profiler output not written. Unexpected exception: ", e)
+    }
+  }
+
+  private def finishWriting(): Unit = {
+    if (profilerDfsDir.isDefined && writing) {
+      try {
+        // shutdown background writer
+        threadpool.shutdown()
+        threadpool.awaitTermination(30, TimeUnit.SECONDS)
+        // flush remaining data
+        writeChunk(true)
+        inputStream.close()
+        outputStream.close()
+      } catch {
+        case _: InterruptedException => Thread.currentThread().interrupt()
+        case e: IOException =>
```

Where does this come from when `writeChunk` swallows all exceptions with `case e: Exception`?